src/plugins/intel_cpu/src/nodes/dft.h

```diff
@@ -19,29 +21,49 @@ class DFT : public Node {
 
     void getSupportedDescriptors() override;
     void initSupportedPrimitiveDescriptors() override;
-    void createPrimitive() override;
     void execute(dnnl::stream strm) override;
     bool created() const override;
 
+    void prepareParams() override;
+
     static bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept;
 
 private:
-    void dftNd(float* output, const std::vector<size_t>& outputStrides) const;
-    void fft(float* data, int64_t dataLength, bool parallelize = false) const;
-    void naiveDFT(float* data, size_t dataLength) const;
+    std::vector<int32_t> getAxes() const;
+    void createJITKernels(bool hasDFT, bool hasFFT);
+
+    void dftNd(float* output,
+               const VectorDims& outputShape,
+               const VectorDims& outputStrides,
+               const std::vector<int32_t>& axes,
+               bool inverse) const;
+
+    float* fft(float* inBuffer,
+               float* outBuffer,
+               int64_t dataLength,
+               bool inverse,
+               bool parallelize = false) const;
+    void naiveDFT(float* data, size_t dataLength, bool inverse) const;
 
-    std::vector<std::pair<float, float>> generateTwiddles(size_t n_complex) const;
+    std::vector<float> generateTwiddlesDFT(size_t n_complex, bool inverse) const;
+    void generateTwiddlesFFT(size_t n_complex, bool inverse);
```

Comment:
M.b. `generateTwiddlesFFT` is more suitable?

Reply:
Done